mne/decoding/search_light.py

```diff
@@ -0,0 +1,332 @@
+        """
+        Fit and transform a series of independent estimators to the dataset.
+
+        Parameters
+        ----------
+        X : array, shape(n_samples, n_features, n_iteration)
+            The training input samples. For each iteration, a clone estimator
+            is fitted independently.
+        y : array, shape(n_samples,)
+            The target values.
+
+        Returns
+        -------
+        y_pred : array, shape(n_samples, n_iterations)
+            Predicted values for each estimator.
+        """
+        return self.fit(X, y).transform(X)
+
+    def fit(self, X, y):
+        """Fit a series of independent estimators to the dataset.
+
+        Parameters
+        ----------
+        X : array, shape(n_samples, n_features, n_iteration)
+            The training input samples. For each iteration, a clone estimator
+            is fitted independently.
+        y : array, shape(n_samples,)
+            The target values.
+
+        Returns
+        -------
+        self : object
+            Return self.
+        """
+        self.estimators_ = list()
+        parallel, p_func, n_jobs = parallel_func(_sl_fit, self.n_jobs)
+        estimators = parallel(
+            p_func(self.base_estimator, split, y)
+            for split in np.array_split(X, n_jobs, axis=2))
+        self.estimators_ = np.concatenate(estimators, 0)
+        return self
+
+    def _transform(self, X, method):
+        """Aux. function to make parallel predictions/transformation"""
+        parallel, p_func, n_jobs = parallel_func(_sl_transform, self.n_jobs)
+        X_splits = np.array_split(X, n_jobs, axis=2)
+        est_splits = np.array_split(self.estimators_, n_jobs)
+        y_pred = parallel(p_func(est, x, method)
+                          for (est, x) in zip(est_splits, X_splits))
+
+        if n_jobs > 1:
+            y_pred = np.concatenate(y_pred, axis=1)
+        else:
+            y_pred = y_pred[0]
+        return y_pred
+
+    def transform(self, X):
+        """Transform the dataset with a series of independent estimators.
+
+        Parameters
+        ----------
+        X : array, shape(n_samples, n_features, n_iteration)
+            The input samples. For each iteration, the corresponding estimator
+            makes a transformation of the data:
+            e.g. [estimators[ii].transform(X[:, :, ii])
+                  for ii in range(n_iteration)]
+
+        Returns
+        -------
+        Xt : array, shape(n_samples, n_iterations)
+            Transformed values generated by each estimator.
+        """
+        if not hasattr(self.base_estimator, 'transform'):
+            ValueError('self.base_estimator does not have `transform` method.')
+        return self._transform(X, 'transform')
+
+    def predict(self, X):
+        """Predict the dataset with a series of independent estimators.
+
+        Parameters
+        ----------
+        X : array, shape(n_samples, n_features, n_iteration)
```

Comment:
Haha ... I meant
`shape(n_samples, n_features, n_estimators)`

Not sure if I fully get what's going on here though.